File: src/main/java/net/dean/jraw/models/attr/Gildable.java
package net.dean.jraw.models.attr;

import net.dean.jraw.models.meta.JsonProperty;

/**
 * Specifies that this model is able to be given reddit gold
 */
public interface Gildable extends JsonAttribute {
    /**
     * Gets the number of times this comment has received reddit gold. If this model was retrieved from the inbox, then
     * this will return 0.
     *
     * @return The number of times this comment has received reddit gold
     */
    @JsonProperty
    Integer getTimesGilded();
}

File: lib/src/main/kotlin/net/dean/jraw/models/Gildable.kt
package net.dean.jraw.models

/**
 * Common interface for models that can receive reddit Gold
 */
interface Gildable {
    /** If the currently-logged-in-user can give reddit Gold to this model */
    val canGild: Boolean

    /** How many times this model has received reddit Gold */
    val gilded: Short
}
